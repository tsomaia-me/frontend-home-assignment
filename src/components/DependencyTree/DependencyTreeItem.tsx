import React from 'react'
import styles from './DependencyTree.module.css'

export interface DependencyTreeItemProps extends React.ComponentProps<'li'> {
  isExpanded?: boolean
  isLoading?: boolean
}

function DependencyTreeItem({ isExpanded, isLoading, children, ...restProps }: DependencyTreeItemProps) {
  return (
    <li {...restProps}>
      <div className={styles.left}>
        <span className="material-icons">{isExpanded ? 'expand_less' : 'expand_more'}</span>
        <span>{children}</span>
      </div>
      {isLoading && (
        <div className={styles.spinner}>
          <span className="material-icons">sync</span>
        </div>
      )}
    </li>
  )
}

export default DependencyTreeItem
