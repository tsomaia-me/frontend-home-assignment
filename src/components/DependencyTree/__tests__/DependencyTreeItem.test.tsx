import React from 'react'
import renderer from 'react-test-renderer'
import DependencyTreeItem from '../DependencyTreeItem'

test('renders DependencyTreeItem correctly', () => {
  const json = renderer.create(<DependencyTreeItem/>).toJSON()
  expect(json).toMatchSnapshot()
})
