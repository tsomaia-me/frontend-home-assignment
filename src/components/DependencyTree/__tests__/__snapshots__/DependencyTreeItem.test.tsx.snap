// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DependencyTreeItem correctly 1`] = `
<li>
  <div
    className="left"
  >
    <span
      className="material-icons"
    >
      expand_more
    </span>
    <span />
  </div>
</li>
`;
